src/components/Tooltip/Tooltip.tsx

```diff
@@ -30,18 +33,46 @@ type Position =
   | "top-left"
   | "top-right";
 
-type Props = {
+export type Props = {
+  /**
+   * Whether the tooltip should adjust to fit in the screen.
+   */
+  autoAdjust?: boolean;
```

Comment:
Instead of having this prop what about defaulting to auto adjusting if the `position` prop is not defined?

Reply:
I think you still want to be able to provide a default position for when it does fit on the screen (e.g. by default you might want it to appear above a button so that it's not covering other content).